examples/fabric-bridge-app/fabric-bridge-common/include/CommissionerControl.h

```diff
@@ -0,0 +1,57 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/clusters/commissioner-control-server/commissioner-control-server.h>
+#include <app/util/af-types.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommissionerControl {
+
+class CommissionerControlDelegate : public Delegate
+{
+public:
+    CHIP_ERROR HandleCommissioningApprovalRequest(const CommissioningApprovalRequest & request) override;
+    CHIP_ERROR ValidateCommissionNodeCommand(NodeId clientNodeId, uint64_t requestId) override;
+    CHIP_ERROR GetCommissioningWindowParams(CommissioningWindowParams & outParams) override;
+    CHIP_ERROR ReverseCommissionNode(const CommissioningWindowParams & params, const Optional<ByteSpan> & ipAddress,
+                                     const Optional<uint16_t> & port) override;
+
+    ~CommissionerControlDelegate() = default;
+
+private:
+    static constexpr size_t kLabelBufferSize = 64;
+
+    uint64_t mRequestId;
+    NodeId mClientNodeId;
+    VendorId mVendorId;
+    uint16_t mProductId;
+    char mLabelBuffer[kLabelBufferSize + 1];
```

Comment:
this is example code. We can use std::string

Reply:
This is used as the backing buffer for label filed (Optional<chip::CharSpan> label;) in the RequestCommissioningApproval command since the original backing buffer will be invalid after returning from emberAfCommissionerControlClusterRequestCommissioningApprovalCallback